source/validate.cpp

```diff
@@ -518,6 +436,109 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Allows for piping of multiple functions which return spv_result_t
+spv_result_t operator||(spv_result_t lhs, spv_result_t rhs) {
```

Comment:
You lose the type of error returned by the function. The validation step returns 0 or 1 for all errors.